### PR TITLE
Rewritten into java

### PR DESCRIPTION
**Pull request by sye8**
Rewritten the filter into java for java application

### CODE REVIEW

File: src/KalmanFilter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * Simple implementation of the Kalman Filter for 1D data, without any dependencies
+ * Originally written in JavaScript by Wouter Bulten
+ * 
+ * Now rewritten into Java
+ * 
+ * @author yesifan
+ * 
+ * @see https://github.com/wouterbulten/kalmanjs
+ *
```
Comment (wouterbulten) [inline src/KalmanFilter.java]: Can you add the license of the project here? See for example the docblock on the js file.

File: src/KalmanFilter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package sye8.utils;
```
Comment (wouterbulten) [inline src/KalmanFilter.java]: Could you rename this package?

File: src/KalmanFilter.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	
+	/**
+	 * 
+	 * @return The last measurement fed into the filter
+	 */
+	public double lastMeasurement(){
+		return this.x;
+	}
+
+	public void setMeasurementNoise(double noise){
```
Comment (wouterbulten) [inline src/KalmanFilter.java]: Please add the docblock here

File: src/KalmanFilter.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		    this.Q = noise;
+	}
+	
+	public void setProcessNoise(double noise){
```
Comment (wouterbulten) [inline src/KalmanFilter.java]: Please add the docblock here

**Review by wouterbulten — CHANGES_REQUESTED** [review]
Hi @ysf199711! Thank you for your pull request. This repo was originally only for the Javascript version of the 1D kalman filter (hence the JS in the name).

However, I get a lot of questions regarding a Java implementation so there is merit in this translation. I would like, however, to keep the main repo/dir clean of any non-js files. Instead, I would like to add a `contrib` folder with contributions.

So, could you move this file to: /contrib/java/KalmanFilter.java? I also added some comments to your code.

After the changes I will merge the branch. Again, thanks for your contributions!

**Comment by sye8** [comment]
Location and comments updated. Package name will be left as (default) as new package name is undecided. Added LGPL v3 license

**Comment by wouterbulten** [comment]
Thanks for the changes! I will merge it in the next few days

**Review by wouterbulten — APPROVED** [review]

**Pull request merged**